File: Android/App/app/src/main/java/com/efalco777/crud_products/presentation/product/mapper/DomainToPresentationProductDetailsMapper.kt
package com.efalco777.crud_products.presentation.product.mapper

import com.efalco777.crud_products.domain.product.model.ProductDetails
import com.efalco777.crud_products.presentation.product.model.PresentationProductDetails

class DomainToPresentationProductDetailsMapper {
    companion object {
        fun map(model: ProductDetails): PresentationProductDetails {
            return PresentationProductDetails(
                id = model.id,
                title = model.title,
                description = model.description,
                category = model.category,
                price = model.price
            )
        }
    }
}

File: Flutter/App/lib/presentation/product/mapper/domain_to_presentation_product_details_mapper.dart
import 'package:products_crud/domain/product/model/product_details.dart';
import 'package:products_crud/presentation/product/model/presentation_product_details.dart';

extension PresentationProductDetailsMapper on ProductDetails {
  PresentationProductDetails toPresentation() {
    return PresentationProductDetails(
      id: id,
      title: title,
      description: description,
      price: price,
      category: category,
      images: images,
    );
  }
}
